controllers/keystoneapi_controller.go

```diff
@@ -712,15 +712,15 @@ func (r *KeystoneAPIReconciler) reconcileNormal(
 	//
 	// check for required OpenStack secret holding passwords for service/admin user and add hash to the vars map
 	//
-	ospSecret, hash, err := oko_secret.GetSecret(ctx, helper, instance.Spec.Secret, instance.Namespace)
+	hash, result, err := oko_secret.VerifySecret(ctx, types.NamespacedName{Name: instance.Spec.Secret, Namespace: instance.Namespace}, []string{"AdminPassword"}, helper.GetClient(), time.Second*10)
 	if err != nil {
 		if k8s_errors.IsNotFound(err) {
```

Comment:
I think since the `IsNotFound` error is handled in `lib-common` [1], this should be checking `result != ctrl.Result{}` instead.

[1] https://github.com/openstack-k8s-operators/lib-common/blob/7fd3da600c9d59fd81e86f32893aaadece9cc856/modules/common/secret/secret.go#L422-L426

Reply:
What I mean by that is have an `else` for the `if err != nil` that checks `result != ctrl.Result{}`.